src/Symfony/Component/DependencyInjection/Container.php

```diff
@@ -178,7 +178,7 @@ public function set($id, $service)
             unset($this->services[$id]);
         }
 
-        if (isset($this->privates[$id])) {
+        if (false !== $randomizedId = array_search($id, $this->privateOriginIds, true)) {
```

Comment:
It's more performand to store a flipped array so you can still use `isset()` which is much faster then a function.
Use usage is to resolve an original id to a private id, so you should target for that 👍 

The Container is called many times during a request, so performance is key (no pun) here.